hknweb/tours/views.py

```diff
@@ -0,0 +1,54 @@
+from django.shortcuts import render, redirect, get_object_or_404
+from django.views import generic
+import datetime
+from django.core.mail import EmailMultiAlternatives
+from django.template.loader import render_to_string
+from django.views.generic.edit import FormView
+from .forms import *
+from django.contrib import messages
+from .models import DepTour
+
+def index(request):
+    tour = DepTour.objects
+
+    context = {
+        'tour': tour,
+    }
+    return render(request, 'tours/index.html', context)
+
+def send_request_email(request, form):
+        subject = 'Department Tour Request'
+        officer_email = 'conhuang@hkn.mu'
+
+        html_content = render_to_string(
+            'tours/request_email.html',
+            {
+                'name' : form.instance.name,
+                'time' : form.instance.desired_time,
+                'date' : form.instance.date,
+                'email': form.instance.email,
+                'phone': form.instance.phone,
+                'comments': form.instance.comments,
+            }
+        )
+        msg = EmailMultiAlternatives(subject, subject,
+                'no-reply@hkn.eecs.berkeley.edu', [officer_email])
+        msg.attach_alternative(html_content, "text/html")
+        msg.send()
+
+def tour(request):
+    form = TourRequest(request.POST or None)
+    if request.method == 'POST':
+        if form.is_valid():
+            tour = form.save(commit=False)
+            tour.save()
+            
+            # send_request_email(request, form)
+
+            messages.success(request, 'Your request has been sent!')
+            return redirect('/tours/')
+        else:
+            print(form.errors)
+            messages.error(request, 'Something went wrong! Your request did not send. Try again, or email deprel@hkn.eecs.berkeley.edu.')
```

Comment:
Do we want to have more specific errors? For example, if the email they entered was invalid, the `TimeField` will error, but the error that's displayed doesn't include anything about the time syntax being incorrect. The user could then assume it's an error on our side, and email Deprel (which is borderline okay, but it's more work for Deprel). 

For example, if I enter this invalid time syntax: 
![hknconnie dept tour pr time submitted invalid form](https://user-images.githubusercontent.com/46059916/76897969-ff1dbc80-6851-11ea-9f3d-3464058fb91e.PNG)
it only displays this on the top: 
![hknconnie dept tour pr time submitted invalid form throws general error](https://user-images.githubusercontent.com/46059916/76898004-0d6bd880-6852-11ea-87f6-165a6bb9fefd.PNG)